src/store/modules/product/actions.ts

```diff
@@ -25,17 +21,53 @@ const actions: ActionTree<ProductState, RootState> = {
     }, '');
 
     if(productIdFilter === '') return;
-
     const resp = await ProductService.fetchProducts({
       "filters": ['productId: (' + productIdFilter + ')']
     })
     if (resp.status === 200 && !hasError(resp)) {
       const products = resp.data.response.docs;
       // Handled empty response in case of failed query
-      if (resp.data) commit(types.PRODUCT_ADD_TO_CACHED_MULTIPLE, { products });
+      if (resp.data) {
+        commit(types.PRODUCT_ADD_TO_CACHED_MULTIPLE, { products });
+      }
     }
-
+    
+    if (productIds.viewIndex === 0) emitter.emit("dismissLoader");
+    // TODO Handle specific error
+    return resp;
+  },
+  async findProduct({ commit }, payload) {
+    let resp;
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    try {
+      resp = await ProductService.fetchProducts({
+        // used sku as we are currently only using sku to search for the product
+        "filters": ['sku: ' + payload.queryString, 'isVirtual: false'],
+        "viewSize": payload.viewSize,
+        "viewIndex": payload.viewIndex
+      })
+      if (resp.status === 200 && resp.data.response?.docs.length > 0 && !hasError(resp)) {
+        const products = resp.data.response.docs;
+        commit(types.PRODUCT_SEARCH_UPDATED, { products });
+        let productIds: any = new Set();
+        products.forEach((item: any) => {
+          productIds.add(item.productId)
+        })
+        productIds = [...productIds]
+        if (productIds.length) {
+          this.dispatch('product/fetchProducts', { productIds });
```

Comment:
Why we are again dispatching the `fetchProducts` action?

Reply:
this was by mistake now it is working fine.